test/test_demos.py

```diff
@@ -0,0 +1,248 @@
+
+
+def longest_word_thread(return_dict):
+    from demo.longest_word import io
+    io.save_to = return_dict
+    io.launch()
+
+
+def sentence_builder_thread(return_dict):
+    from demo.sentence_builder import io
+    io.save_to = return_dict
+    io.launch()
+
+
+class TestDemo(unittest.TestCase):
+    def test_diff_texts(self):
+        manager = multiprocessing.Manager()
+        return_dict = manager.dict()
+        self.i_thread = multiprocessing.Process(target=diff_texts_thread,
+                                         args=(return_dict,))
+        self.i_thread.start()
+        while not return_dict:
+            time.sleep(0.2)
+
+        URL = LOCAL_HOST.format(return_dict["port"])
+        wait_for_url(URL)
+
+        driver = webdriver.Chrome()
+        driver.set_window_size(1200, 800)
+        driver.get(URL)
+        timeout = 10
+        elem = WebDriverWait(driver, timeout).until(
+            EC.presence_of_element_located((By.CSS_SELECTOR,
+                                            ".input_interface[interface_id='0'] .input_text"))
+        )
+        elem.clear()
+        elem.send_keys("Want to see a magic trick?")
+        elem = WebDriverWait(driver, timeout).until(
+            EC.presence_of_element_located((By.CSS_SELECTOR,
+                                            ".input_interface[interface_id='1'] .input_text"))
+        )
+        elem.clear()
+        elem.send_keys("Let's go see a magic trick!")
+        elem = WebDriverWait(driver, timeout).until(
+            EC.presence_of_element_located((By.CSS_SELECTOR,
+                                            ".submit"))
+        )
+        elem.click()
+        elem = WebDriverWait(driver, timeout).until(
+            EC.presence_of_element_located((By.CSS_SELECTOR,
+                                            ".output_interface[interface_id='2'] .output_text"))
+        )
+        while elem.text == "":
+            time.sleep(0.2)
+
+        self.assertEqual(elem.text, "LeWant's tgo see a magic trick?!")
+        golden_img = os.path.join(current_dir, GOLDEN_PATH.format(
+            "diff_texts", "magic_trick"))
+        tmp = os.path.join(current_dir, "test/tmp/{}.png".format(
+            random.getrandbits(32)))
+        hide_latency(driver)
+        driver.save_screenshot(tmp)
+        driver.close()
+        self.assertIsNone(compare_images(tmp, golden_img, TOLERANCE))
+        os.remove(tmp)
+
+    def test_image_mod(self):
+        manager = multiprocessing.Manager()
+        return_dict = manager.dict()
+        self.i_thread = multiprocessing.Process(target=image_mod_thread,
+                                                args=(return_dict,))
+        self.i_thread.start()
+        while not return_dict:
+            time.sleep(0.2)
+
+        URL = LOCAL_HOST.format(return_dict["port"])
+        wait_for_url(URL)
+
+        driver = webdriver.Chrome()
+        driver.set_window_size(1200, 800)
+        driver.get(URL)
+        timeout = 10
```

Comment:
seems like everything up till here is common between tests, move to setUp?

Reply:
Couldn't figure out how to pass a parameter to setUp for the specific demo thread, so opted to refactor with a regular function. See https://github.com/gradio-app/gradio/pull/72/commits/200d7031b5c8713c3a6b1914e67fd9aa22e633a7 and https://github.com/gradio-app/gradio/pull/72/commits/200d7031b5c8713c3a6b1914e67fd9aa22e633a7